torch/csrc/api/include/torch/nn/options/activation.h

```diff
@@ -82,7 +82,23 @@ struct TORCH_API SoftmaxOptions {
   TORCH_ARG(int64_t, dim);
 };
 
-TORCH_NN_FUNCTIONAL_USE_MODULE_OPTIONS(Softmax, SoftmaxFuncOptions)
+// ============================================================================
+
+namespace functional {
+
+struct TORCH_API SoftmaxFuncOptions {
+  SoftmaxFuncOptions(int64_t dim);
+
+  /// Dimension along which Softmax will be computed.
+  TORCH_ARG(int64_t, dim);
```

Comment:
I believe if we don't pass a non-None `dim` value to Softmax python module and functional, a warning will be shown from `_get_softmax_dim`: 
https://github.com/pytorch/pytorch/blob/c49b324cbf17272d02f3811f517fe82106ce77d4/torch/nn/functional.py#L1193-L1194
https://github.com/pytorch/pytorch/blob/c49b324cbf17272d02f3811f517fe82106ce77d4/torch/nn/functional.py#L1166-L1174
Since `dim=None` is already deprecated in Python frontend, for C++ frontend we should require `dim` to be provided.